disable peephole optimizations that require alias db

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37105 [JIT] remove dominated guards of functional values
* #37012 optimize is_float_point calls
* #36953 Add support for generic list constants
* #36829 [JIT] Add Lower Tuples Call & Run remove mutation after list unrolling
* #36828 add post unroll optimizations
* #36758 Add size peephole optimziation
* **#36757 disable peephole optimizations that require alias db**

replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

Commit:
disable peephole optimizations that require alias db

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

[ghstack-poisoned]

Krovatkin reviewed (approved):
:shipit: i'm surprised there are no changes to test files. This PR broke no tests?

eellison commented:
@krovatkin there were tests that the peephole didn't incorrectly run but no tests that it did run in the right scenarios

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]

Commit:
Update on "disable peephole optimizations that require alias db"


replacing x + 0 with x isn't that much of a speedup, and is an optimization also duplicated at the Tensor Expr level. Constructing an alias db is costly and it's not worth rebuilding an alias db each time we optimize out x + 0.

Differential Revision: [D21160757](https://our.internmc.facebook.com/intern/diff/D21160757)

[ghstack-poisoned]
